SCons/Scanner/Python.py

```diff
@@ -92,10 +126,14 @@ def scan(node, env, path=()):
         # if the same header is included many times.
         node.includes = list(map(SCons.Util.silent_intern, includes))
 
-    # XXX TODO: Sort?
     nodes = []
     if callable(path):
         path = path()
+
+    # If there are no paths, there is no point in parsing includes in the loop.
+    if not path:
```

Comment:
Is this correct? what if the modules are located in the same directory as the file?

Reply:
Yeah good point. I'll need to move it below and change it to:

if not search_paths:

Reply:
Any updates?

Reply:
Oops, sorry! Just pushed an update with a fix.

Separately, do you want me to merge master into this branch to bring it up-to-date? Wasn't sure if it would make the diff more complicated or if it would be transparent.

Reply:
Nevermind, I figured out how to merge master in this branch, so I have done that.